forta-bots/DAI-Price-Monitor/src/agent.ts

```diff
@@ -0,0 +1,101 @@
+import BigNumber from "bignumber.js";
+import { providers } from "ethers";
+import {
+  Finding,
+  getEthersProvider,
+  HandleTransaction,
+  Initialize,
+  LogDescription,
+  TransactionEvent,
+} from "forta-agent";
+import Fetcher from "./fetcher";
+import NetworkData, { NETWORK_MAP } from "./network";
+import NetworkManager from "./network";
+import { SWAP_EVENT, TOKEN_EXCHANGE_EVENT } from "./abi";
+import {
+  createFinding,
+  calculateUniswapPrice,
+  calculateCurvePrice,
+  uniswapPairCreate2,
+  SPREAD_THRESHOLD,
+} from "./utils";
+
+const networkManager: NetworkData = new NetworkManager(NETWORK_MAP);
+
+export const initialize =
+  (provider: providers.Provider): Initialize =>
+  async () => {
+    const { chainId } = await provider.getNetwork();
+    networkManager.setNetwork(chainId);
+  };
+
+export const provideHandleTransaction = (
+  data: NetworkData,
+  fetcher: Fetcher,
+  spreadThreshold: BigNumber
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+    const block: number = txEvent.blockNumber;
+    const uniswapLogs: LogDescription[] = txEvent.filterLog(SWAP_EVENT);
+
+    await Promise.all(
+      uniswapLogs.map(async (log: LogDescription) => {
+        const [valid, token0, token1, fee] = await fetcher.getPoolData(block, log.address);
+
+        if (valid && log.address === uniswapPairCreate2(token0, token1, fee, data)) {
+          if (
+            data.DAI < data.USDC ? token0 == data.DAI && token1 == data.USDC : token0 == data.USDC && token1 == data.DAI
+          ) {
```

Comment:
Some things you can do here:
You know DAI address and USDC address, so you can decide which one is `token0` & `token1` (with the check in line 48) so you can create this 2 variables somewhere to avoid checking it all the time (probably inside the networkManager). By doing this you can keep the same check you have in L46, but notice that you are using the `token0` & `token1` you are specting instead of fetching it.